examples/polypharmacy_gripnet/model.py

```diff
@@ -1,86 +1,131 @@
+from typing import Tuple
+
 import numpy as np
+import pytorch_lightning as pl
 import torch
-from torch import nn
-from utils import negative_sampling
+from utils import auprc_auroc_ap, EPS, typed_negative_sampling
+from yacs.config import CfgNode
 
-from kale.embed.gripnet import TypicalGripNetEncoder
+from kale.embed.gripnet import GripNet
+from kale.prepdata.supergraph_construct import SuperGraph
 
 
-class MultiRelaInnerProductDecoder(nn.Module):
-    """
-    Build `DistMult
-    <https://arxiv.org/abs/1412.6575>`_ factorization as GripNet decoder in PoSE dataset.
+class MultiRelaInnerProductDecoder(torch.nn.Module):
```

Comment:
DistMult is highly reusable and should go into the core API, likely `kale.predict`, and tested.